modules/standard/FileSystem.chpl

```diff
@@ -498,8 +498,9 @@ proc locale.cwd(out error: syserr): string {
     if (error != ENOERR) {
       ret = "";
     } else {
-      // This version of toString steals its operand.  No need to free.
-      ret = toString(tmp);
+      var tmp_len = tmp.length;
+      ret = new string(tmp:c_ptr(uint(8)), tmp_len, tmp_len,
+                       owned=true, needToCopy=false);
```

Comment:
ah, here it is. I'd rather this call a function with Steal in the name, or that the relevant constructor have a buffer argument called StealBuffer e.g.